python/tank/util/filesystem.py

```diff
@@ -451,3 +453,50 @@ def get_unused_path(base_path):
         if not os.path.exists(path):
             break
     return path
+
+
+@with_cleared_umask
+@contextmanager
+def auto_created_yml(path):
```

Comment:
Moved this here since it was needed in multiple locations. I think it makes sense??